apps/files/src/components/Collaborators/AdditionalPermissions.vue

```diff
@@ -0,0 +1,64 @@
+<template>
+  <oc-grid gutter="small">
+    <label v-for="permission in permissions" :key="permission.name">
+      <oc-checkbox
+        class="uk-margin-xsmall-right"
+        v-model="permission.value"
+        @change="permissionChecked"
+      />
+      {{ permission.description }}
+    </label>
+  </oc-grid>
+</template>
+
+<script>
+import { mapGetters } from 'vuex'
+
+const filter = require('filter-obj')
+
+export default {
+  name: 'AdditionalPermissions',
+  props: {
+    selectedRole: {
+      type: Object,
+      required: true
+    },
+    collaborator: {
+      type: Object,
+      required: false
+    }
+  },
+  computed: {
+    ...mapGetters('Files', ['highlightedFile']),
+
+    permissions () {
+      const permissions = this.selectedRole.additionalPermissions
```

Comment:
how about making the component generic and not rely on selectedRole and collaborators and only simply have properties "availablePermissions"  and "selectedPermissions".